source/core/slang-shader-cache-index.h

```diff
@@ -0,0 +1,74 @@
+// slang-shader-cache-index.h
+#pragma once
+#include "../../slang.h"
+#include "../../slang-com-ptr.h"
+
+#include "slang-string.h"
+#include "slang-dictionary.h"
+#include "slang-linked-list.h"
+
+namespace Slang
+{
+
+struct ShaderCacheEntry
+{
+    slang::Digest dependencyBasedDigest;
+    slang::Digest astBasedDigest;
+};
+
+class ShaderCacheIndex
+{
+public:
+    ShaderCacheIndex(SlangInt size, String path, ISlangFileSystem* fileSystem, String filename);
```

Comment:
Use more descriptive parameter names. `size`->`entryCountLimit`, `filename`->`indexFileName`.
You can wrap these parameters in a `struct ShaderCacheIndexDesc`, and make it the only parameter of the constructor. The class can hold an member of `ShaderCacheIndexDesc desc` so you don't have to assign them manually one by one.

Reply:
There is already a shader cache Desc in slang.h so we can just reuse that.

Reply:
I'll change this again in the integration PR where the ShaderCacheDesc struct is introduced